tests/hf/test_integrals.py

```diff
@@ -47,3 +57,208 @@ def test_gaussian_norm(l, alpha, n):
+            pnp.array([3.42525091]),
+            pnp.array([0.0]),
+        ),
+        (
+            (1, 0, 0),
+            (0, 0, 1),
+            pnp.array([0.0, 0.0, 0.0]),
+            pnp.array([0.0, 0.0, 0.0]),
+            pnp.array([6.46480325]),
+            pnp.array([6.46480325]),
+            pnp.array([0.0]),
+        ),
+    ],
+)
+def test_gaussian_overlap(la, lb, ra, rb, alpha, beta, o):
+    r"""Test that gaussian overlap function returns a correct value."""
+    assert np.allclose(gaussian_overlap(la, lb, ra, rb, alpha, beta), o)
+
+
+@pytest.mark.parametrize(
+    ("symbols", "geometry", "alpha", "coef", "r", "o_ref"),
+    [
+        (
+            ["H", "H"],
+            pnp.array([[0.0, 0.0, 0.0], [0.0, 0.0, 20.0]], requires_grad=False),
+            pnp.array(
+                [[3.42525091, 0.62391373, 0.1688554], [3.42525091, 0.62391373, 0.1688554]],
+                requires_grad=False,
+            ),
+            pnp.array(
+                [[0.15432897, 0.53532814, 0.44463454], [0.15432897, 0.53532814, 0.44463454]],
+                requires_grad=True,
+            ),
+            pnp.array([[0.0, 0.0, 0.0], [0.0, 0.0, 20.0]], requires_grad=True),
+            pnp.array([0.0]),
+        ),
+        (
+            ["H", "H"],
+            pnp.array([[0.0, 0.0, 0.0], [0.0, 0.0, 0.0]], requires_grad=False),
+            pnp.array(
+                [[3.42525091, 0.62391373, 0.1688554], [3.42525091, 0.62391373, 0.1688554]],
+                requires_grad=False,
+            ),
+            pnp.array(
+                [[0.15432897, 0.53532814, 0.44463454], [0.15432897, 0.53532814, 0.44463454]],
+                requires_grad=False,
+            ),
+            pnp.array([[0.0, 0.0, 0.0], [0.0, 0.0, 0.0]], requires_grad=True),
+            pnp.array([1.0]),
+        ),
+        (
+            ["H", "H"],
+            pnp.array([[0.0, 0.0, 0.0], [0.0, 0.0, 0.0]], requires_grad=True),
+            pnp.array(
+                [[3.42525091, 0.62391373, 0.1688554], [3.42525091, 0.62391373, 0.1688554]],
+                requires_grad=True,
+            ),
+            pnp.array(
+                [[0.15432897, 0.53532814, 0.44463454], [0.15432897, 0.53532814, 0.44463454]],
+                requires_grad=True,
+            ),
+            pnp.array([[0.0, 0.0, 0.0], [0.0, 0.0, 0.0]], requires_grad=True),
+            pnp.array([1.0]),
+        ),
+    ],
+)
+def test_generate_overlap(symbols, geometry, alpha, coef, r, o_ref):
+    r"""Test that generate_overlap function returns a correct value for the overlap integral."""
+    mol = Molecule(symbols, geometry)
+    basis_a = mol.basis_set[0]
+    basis_b = mol.basis_set[1]
+    args = [p for p in [alpha, coef, r] if p.requires_grad]
+
+    o = generate_overlap(basis_a, basis_b)(*args)
+    assert np.allclose(o, o_ref)
+
+
+@pytest.mark.parametrize(
+    ("symbols", "geometry", "alpha", "coeff", "g_ref"),
+    [
+        # g_ref computed manually with finite diff
```

Comment:
Perhaps it's worth having the finite-diff computation itself in the test file? This is minor though

Reply:
Done.